helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java

```diff
@@ -1705,4 +1735,26 @@ private static void verifyToken(final String token, final Job job)
     }
   }
 
+  private String getPreviousJobStates(final JobId jobId, final String host) {
```

Comment:
This seems ugly. And which class should such a function live (this class is already too big).